src/plugins/data/public/ui/query_string_input/language_switcher.tsx

```diff
@@ -42,15 +43,32 @@ import {
 } from '@elastic/eui';
 import { FormattedMessage } from '@osd/i18n/react';
 import React, { useState } from 'react';
+import { Observable } from 'rxjs';
+import { useObservable } from 'react-use';
 import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
 
 interface Props {
   language: string;
   onSelectLanguage: (newLanguage: string) => void;
   anchorPosition?: PopoverAnchorPosition;
+  currentApp$?: Observable<string | undefined>;
+  useNewQuerySelector?: boolean;
 }
 
 export function QueryLanguageSwitcher(props: Props) {
+  const dataExplorerOptions = [
+    {
+      label: 'DQL',
+    },
+    {
+      label: 'lucene',
+    },
+    {
+      label: 'PPL',
+    },
+  ];
+  const [selectedLanguage, setSelectedLanguage] = useState([dataExplorerOptions[0]]);
+
   const osdDQLDocs = useOpenSearchDashboards().services.docLinks?.links.opensearchDashboards.dql
```

Comment:
Maybe we can access `application?.currentAppId$;` here directly instead of passing `currentApp$` all the way down to `QueryLanguageSwitcher`?